### PR TITLE
#175636847 Add routes for getting and updating profile information

### PR DESCRIPTION
**Pull request by hervebutera**
#### What does this PR do?
Adds functionality for getting user information by id.
Adds functionality for updating user information.
#### Description of Task to be completed?
- When the user is logged in, he/she can get his/her profile
information by id.
- The logged-in user also is able to update his profile information.
- The user is also able to change his profile picture which is uploaded on Cloudinary.
#### How should this be manually tested?
- clone the repo by `git clone https://github.com/atlp-rwanda/tech-strikers-bn-backend.git`
- run `cd tech-strikers-bn-backend`
- run `git checkout ft-profile-page-settings-175636847`
- run `npm install`
- add `CLOUDINARY_NAME=` in .env file and corresponding value of your cloudinary account for testing.
- add `CLOUDINARY_API_KEY=` in .env file and corresponding value of your cloudinary account for testing.
- add `CLOUDINARY_API_SECRET=` in .env file and corresponding value of your cloudinary account for testing.
- In your cloudinary account settings, go to upload settings, create an `upload preset` with mode `Unsigned`.
- Also in `src/controllers/user.controller.js`, add the created upload preset name in the second parameter 
of  `uploadProfilePic` function.
- run `npm run dev`
- use postman to test `GET  http://localhost:3000/api/user/<any user id>`
-use postman to test  `PUT http://localhost:3000/api/user/<any user id>` 
#### Any background context you want to provide?
Cloudinary offers limited storage GB for a free account so every developer has to use his cloudinary account for
development and testing purposes.
Also a default value of a cloudinary image has been added in the `profilePicture` column for every user.
#### What are the relevant pivotal tracker stories?
[#175636847](https://www.pivotaltracker.com/story/show/175636847)
#### Screenshots (if appropriate)
- Postman screenshot of `GET` route for user table.
Capture1

- Postman screenshot of unsuccessful `PUT` operation when `username` has been taken
Capture2

- Postman screenshot of successful `PUT` route (updating user information)for user 
Capture

### CODE REVIEW

**Review by tytyne — CHANGES_REQUESTED** [review]
I thought changing profile, we had to add picture  and other information related to the user!

File: src/controllers/user.controller.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    if (dbResponse === 'true') {
+        res.status(200).send({"message" : "Profile information updated successfully"})
+    } else if (dbResponse === "false") {
+        res.status(400).send({"message" : "Unable to update your profile"})
+    }
+}
+
+
+
+export default { getUserInfo, upDateUser };
```
Comment (Duncanwar) [inline src/controllers/user.controller.js]: Can you transform your functions into a class?
For readability and solid principles

File: src/services/user.service.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          
+        return true
+        }    
+    catch {(error) => {
+        console.log(error)
+        return false
+    }}
+} 
+
+export default { retrieveUser, upDateUserInfo };
```
Comment (Duncanwar) [inline src/services/user.service.js]: We can use a class

File: src/routes/api/users.js [hunk 3]
```diff
@@ -1,29 +1,10 @@
...
-});
-
-router.post("/user", (req, res, next) => {
-  res.status(200).json({
-    message: "successfully sent"
-  });
-});
+router.put("/user/:userId", upDateUser);
 
 export default router;
```
Comment (Duncanwar) [inline src/routes/api/users.js]: You don't have to remove the above lines.

**Review by Duncanwar — COMMENTED** [review]
Implement those feedback

**Comment by hervebutera** [comment]
> I thought changing profile, we had to add a picture and other information related to the user!

Now user information and a profile picture included can be updated.

**Review by tytyne — APPROVED** [review]

**Review by Duncanwar — APPROVED** [review]

**Pull request merged**